stripe/src/main/java/com/stripe/android/paymentsheet/BaseAddCardFragment.kt

```diff
@@ -277,6 +280,33 @@ internal abstract class BaseAddCardFragment(
                 )
             }
         cardMultilineWidget.postalCodeErrorListener = null
+
+        billingAddressView.postalCodeViewListener =
+            object : BillingAddressView.PostalCodeViewListener {
+                override fun onLosingFocus(country: Country?, isPostalValid: Boolean) {
+                    if (isPostalValid) {
+                        billingErrors.text = null
+                        billingErrors.isVisible = false
+                    } else {
+                        if (country == null || country.code == "US") {
+                            billingErrors.text = getString(R.string.address_zip_invalid)
+                        } else {
+                            billingErrors.text = getString(R.string.address_postal_code_invalid)
+                        }
+                        billingErrors.isVisible = true
+                    }
+                }
+
+                override fun onGainingFocus(country: Country?, isPostalValid: Boolean) {
+                    // Always hide error field when user starts editing postal code
+                    billingErrors.isVisible = false
+                }
+
+                override fun onCountryChanged(country: Country?, isPostalValid: Boolean) {
+                    billingErrors.text = null
+                    billingErrors.isVisible = false
```

Comment:
If the country changes, can there still be a postal/zip error, right?  At first glance, it seems like onCountryChanged should look if the postal is valid.

Reply:
Yes I actually did that initially, but there is an edge case with the following steps:
1) Input some invalid zip(say "abc") in non-us country, the field is non-red since it's not  being validated. 
2) Then we change the country to US, if check validation upon country here, the field will be turn red. 
3) Now since we only validate it on LosingFocus or CountryChanged, the field will keep being red even though user has deleted "abc" and input correct format, so they will see a valid ZIP code in red and "Your ZIP code is Invalid" error message.
[here](https://drive.google.com/file/d/1LGt_4Hl1fR_JNqXIcaDRVGoadnYiMco8/view?usp=sharing) is a recording

Note this behavior happens not only for the error messages just added, but also happens on `BillingAddressView#postalCodeView` (the text will be red even though user input correct ZIP, to resolve that we'll need to add logic to trigger on keystroke change in the text field and does partial postal code matching on top. 

I think it's a pretty niche case and is beyond the scope of this PR(just adding error field), probably it's better for now to just clear them all upon country update, if we need to fix this we can do it in a separate ticket.

Reply:
I think this would require setting the error text color for the postal field in BillingAddressView#createAddress which is called on each text change?

Can you create a JIRA ticket for this?

Reply:
Yes, basicly `BillingAddressView` needs to expose a yes/no whether postal is valid on each character input, created [this](https://jira.corp.stripe.com/browse/MOBILESDK-197) to track this